### PR TITLE
Create Java Solution for Do You Know You're ABC's

### PR DESCRIPTION
**Pull request by MrinallU**

### CODE REVIEW

**Comment by MrinallU** [comment]
@bqi343 Should be good for review.

**Comment by thecodingwizard** [comment]
Isn't the official solution a lot simpler than this solution? You don't have to generate every permutation right?

File: solutions/USACO dec20-abc-bronze.mdx [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+
+<LanguageSection>
+
+<JavaSection>
+
+```java
+ import java.io.*;
+ import java.util.*;
+
+ public class abc {
```
Comment (frodakcin) [inline solutions/USACO dec20-abc-bronze.mdx:21]:
```suggestion
public class abc {
```

Delete the extra space in front of lines. For example, the suggestion above (it's also preferable if you switch to tabs).

File: solutions/USACO dec20-abc-bronze.mdx [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+          return true;
+    }
+
+    // Given the a permutation check if the values fufill the condition given in the problem statement.
+    public static boolean valid(int[] arr){ 
+         int a = arr[0]; int b = arr[1]; int c = arr[2];
+         int aPlusB = arr[3]; int bPlusC = arr[4]; int cPlusA = arr[5]; 
+         int totalVal = arr[6];
+
+         return totalVal == (a + b + c) && aPlusB == (a + b) && bPlusC == (b + c) && cPlusA == (c + a);
```
Comment (frodakcin) [inline solutions/USACO dec20-abc-bronze.mdx:78]: Line is too long. Easiest fix is to add and endline after each `&&`

File: solutions/USACO dec20-abc-bronze.mdx [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+    }
+     
+    public static void main(String args[]) {
```
Comment (frodakcin) [inline solutions/USACO dec20-abc-bronze.mdx:81]: Inconsistently adding/excluding space between `()` and `{`

**Review by frodakcin — CHANGES_REQUESTED** [review]
Formatting issues

**Comment by frodakcin** [comment]
Nathan's point still stands though (IDK the problem)

Also, is passing and returning arrays in Java good practice?

**Comment by MrinallU** [comment]
@frodakcin Yes, the problem can be solved with an easier method, but I don't think there is wrong with showing another way to solve it.

**Comment by MrinallU** [comment]
> Formatting issues

Should be good.

File: solutions/USACO dec20-abc-bronze.mdx [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+    public static boolean valid(int[] arr){ 
+         int a = arr[0]; int b = arr[1]; int c = arr[2];
+         int aPlusB = arr[3]; int bPlusC = arr[4]; int cPlusA = arr[5]; 
+         int totalVal = arr[6];
+         
+         
+         return (totalVal == (a + b + c) && 
+         aPlusB == (a + b) && 
+         bPlusC == (b + c) && 
+         cPlusA == (c + a))
```
Comment (frodakcin) [inline solutions/USACO dec20-abc-bronze.mdx:82]: semicolon??!

Also, add another tab before each of these 3 lines (like hanging indent for multiline paragraphs)

**Review by frodakcin — CHANGES_REQUESTED** [review]
I think you should switch to tabs to make it easier. There are some really weird spacing issues throughout the code.

File: solutions/USACO dec20-abc-bronze.mdx (hunk 4 above)
Comment (MrinallU) [reply solutions/USACO dec20-abc-bronze.mdx:82]: Got it.

**Comment by MrinallU** [comment]
@frodakcin Next time I add a solution I'll definitely add a space.

**Review by thecodingwizard — CHANGES_REQUESTED** [review]
- change indentation to tabs
- Should probably add a note before implementation that specifies that the below solution is strongly discouraged, and to refer to the editorial instead
- Might also want to briefly explain what the code is doing?

File: solutions/USACO dec20-abc-bronze.mdx [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+<LanguageSection>
+
+<JavaSection>
+
+```java
+ import java.io.*;
+ import java.util.*;
+
+public class abc{
+      public static int[] swap(int d[], int l, int r){
```
Comment (frodakcin) [inline solutions/USACO dec20-abc-bronze.mdx:22]: 6 space indent?

File: solutions/USACO dec20-abc-bronze.mdx [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+           int temp = d[l];
```
Comment (frodakcin) [inline solutions/USACO dec20-abc-bronze.mdx:23]: 6 + 5 space indent?

**Review by frodakcin — CHANGES_REQUESTED** [review]
There are 6, 5, and 4 space indents in this code.

(What text editor are you using and why doesn't it automatically fix this?!)

**Comment by frodakcin** [comment]
Stalling + official solution is more insightful and just better

**Pull request closed without merging**